extensions_userprovided/raspiBackupDiscordWrapper.sh

```diff
@@ -0,0 +1,241 @@
+         [[ -z $PATH ]] && export PATH=$p || export PATH="$p:$PATH"
+      fi
+   done
+fi
+
+
+function readVars() {
+	if [[ -f /tmp/raspiBackup.vars ]]; then
+		source /tmp/raspiBackup.vars						# retrieve some variables from raspiBackup for further processing
+# now following variables are available for further backup processing
+# BACKUP_TARGETDIR refers to the backupdirectory just created
+# BACKUP_TARGETFILE refers to the dd backup file just created
+	else
+		echo "/tmp/raspiBackup.vars not found"
+		exit 42
+	fi
+}
+
+
+
+#### DISCORD SECTION
+
+# read config file
+source /usr/local/etc/discordWrapper.conf
+
+###########################################
+### EXPECTED CONFIG FILE CONTENT
+#
+# URL of the Discord WebHook where to send the message. Adapt to your case
+# CONFIG_DISCORD_URL="https://discord.com/api/webhooks/1234567890.../abcdef...."
+#
+# Possible Colors to be used for Discord Embed message.
+# COLOR_BLUE=255
+# COLOR_RED=16711680
+# COLOR_GREEN=65280
+#
+#
+# Color to be used depending on the raspiBackup execution result: success or failure
+# CONFIG_COLOR_DEFAULT=$COLOR_BLUE
+# CONFIG_COLOR_SUCCESS=$COLOR_GREEN
+# CONFIG_COLOR_FAILURE=$COLOR_RED
+#
+#
+##########################################
+
+DISC_CONTENT=""
+DISC_TITLE=""
+DISC_DESC=""
+DISC_COLOR=$CONFIG_COLOR_DEFAULT
+
+###########################################
+### Default JSON message for Discord notification with a colored embed
+###########################################
+DEFAULTJSON=$(cat<<EOS
+{"content": "DISC CONTENT",
+  "embeds": [{
+    "title": "DISC TITLE",
+    "description": "DISC DESC",
+    "color": "255"
+  }]
+}
+EOS
+)
+
+
+# holds the JSON data to post
+DISCORD_JSON=""
+###########################################
+### Function to construct the JSON data to be posted
+# It requires jq to be installed
+# there is not parameter
+# it uses the following script variable:
+# DISC_CONTENT
+# DISC_TITLE
+# DISC_DESC
+# DISC_COLOR
+#
+# these variables are set in function send_discord
+###########################################
+function build_json() {
+        DISCORD_JSON=$(jq \
+                        --arg cont "$DISC_CONTENT" \
+                        --arg title "$DISC_TITLE" \
+                        --arg color "$DISC_COLOR" \
+                        --arg desc "$DISC_DESC" \
+                       '.content=$cont | .embeds[0].title=$title | .embeds[0].description=$desc | .embeds[0].color=$color' <<< "$DEFAULTJSON")
+}
+###########################################
+### Function to set variables for JSON message, and calls JSON construction function: build_json
+# it then send the message to discord using CURL
+# $1 contains the raspiBackup return code in order to adapt the message embed's color
+###########################################
+function send_discord() { # $1:status of the operation, 0=) success, not 0 is failure
+        local discord_color=$CONFIG_COLOR_DEFAULT
+        if [[ $1 -ne  0 ]]; then
```

Comment:
I prefer to use `(( $1 != 0 ))` which is more readable

Reply:
Correct, we all have some habits ;)

Reply:
done